std/fmt.zig

```diff
@@ -39,66 +66,179 @@ pub fn format(
+                    options.width.? += c - '0';
+                },
+                '.' => {
+                    state = .FormatPrecision;
+                },
+                '}' => {
+                    const arg_to_print = comptime nextArg(&used_pos_args, maybe_pos_arg, &next_arg);
+
+                    try formatType(
+                        args[arg_to_print],
+                        fmt[specifier_start..specifier_end],
+                        options,
+                        context,
+                        Errors,
+                        output,
+                        default_max_depth,
+                    );
+                    state = .Start;
+                    start_index = i + 1;
+                },
+                else => {
+                    @compileError("Unexpected character in width value: " ++ [_]u8{c});
+                },
+            },
+            .FormatPrecision => switch (c) {
+                '0'...'9' => {
+                    if (options.precision == null) {
+                        options.precision = 0;
+                    }
+
+                    options.precision.? *= 10;
+                    options.precision.? += c - '0';
+                },
+                '}' => {
+                    const arg_to_print = comptime nextArg(&used_pos_args, maybe_pos_arg, &next_arg);
+
+                    try formatType(
+                        args[arg_to_print],
+                        fmt[specifier_start..specifier_end],
+                        options,
+                        context,
+                        Errors,
+                        output,
+                        default_max_depth,
+                    );
+                    state = .Start;
+                    start_index = i + 1;
+                },
+                else => {
+                    @compileError("Unexpected character in precision value: " ++ [_]u8{c});
+                },
+            },
             .Pointer => switch (c) {
                 '}' => {
-                    try output(context, @typeName(@typeOf(args[next_arg]).Child));
+                    const arg_to_print = comptime nextArg(&used_pos_args, maybe_pos_arg, &next_arg);
+
+                    try output(context, @typeName(@typeOf(args[arg_to_print]).Child));
                     try output(context, "@");
-                    try formatInt(@ptrToInt(args[next_arg]), 16, false, 0, context, Errors, output);
-                    next_arg += 1;
-                    state = State.Start;
+                    try formatInt(@ptrToInt(args[arg_to_print]), 16, false, 0, context, Errors, output);
+
+                    state = .Start;
                     start_index = i + 1;
                 },
                 else => @compileError("Unexpected format character after '*'"),
             },
         }
     }
     comptime {
-        if (args.len != next_arg) {
+        // All arguments must have been printed but we allow mixing positional and fixed to achieve this.
```

Comment:
This is to handle intermixing positionals and non-positionals where by themselves we fail to iterate through the whole set.